### PR TITLE
Update mail.py to allow html content

### PR DESCRIPTION
**Pull request by adematte**
Adding parameter subtype to allow html based content. The default behavior of text/plain has been preserved.

### CODE REVIEW

**Comment by gregdek** [comment]
@dagwieers Please review; if you give +1 we will merge

**Comment by bcoca** [comment]
needs rebase

File: notification/mail.py [hunk 1]
```diff
@@ -111,12 +111,16 @@
       - The character set of email being sent
     default: 'us-ascii'
     required: false
+  subtype:
```
Comment (bcoca) [inline notification/mail.py:113]: mimetype might be a better name for this option

Comment (adematte) [inline notification/mail.py:113]: true! i'll rebase and make the necessary changes when I get the chance to.

Comment (dagwieers) [inline notification/mail.py:113]: Well, it is not really a mimetype you have to provide, maybe mimesubtype would be preferable ?
I prefer subtype over mimetype (as it is more correct), cfr. https://docs.python.org/2/library/mimetypes.html

Whatever the decision is, I support this change. +1

**Comment by bcoca** [comment]
we can split the alias? I'm just guessing people will not be as familiar with the subtype as with the mimetype name for this feature.

**Comment by dagwieers** [comment]
Why not just make it clear in the documentation that subtype, is the "minor" part of a "text" MIME type ? (which isn't very clear now from the description)

At least that way it does not pretend to expect a full mimetype, and anyone who does not know what is to be expected, will read from the documentation what it is. Besides, now only 'plain' and 'html' are allowed, or are any of the other "text" MIME types a possibility as well ? e.g. text/richtext, text/sgml, text/xml or text/x-uuencode

**Comment by bcoca** [comment]
@dagwieers agreed, documenting it clearly seems to be the best solution

**Comment by adematte** [comment]
rebased and improved description of the parameter. Good to go?

**Comment by gregdek** [comment]
Back to @dagwieers for re-review.

**Comment by dagwieers** [comment]
I support this change. Simple and concise. +1

**Comment by gregdek** [comment]
Merging. Thanks @dagwieers for the review, and thanks @adematte for the PR.

**Comment by resmo** [comment]
version_added of the new arg is missing...

**Comment by abadger** [comment]
Good spot @resmo!  Added version_added: 955bf92ff892a7359a045e1ddb3b29b7809a230b

**Pull request merged**